### ISSUE TITLE
Interceptor BCC addresses are not working

### ISSUE THREAD

**Issue by xtagon** [renz45/mandrill_mailer#80]
Hi,

I would like to use the `interceptor_params` to create a BCC recipient (or more than one BCC recipient) for all messages sent, but I would like the original recipient to still receive the message. If I set `interceptor_params = {to: {email: "copy@example.com",  type: "bcc"}}` then *only* the copy is sent, not the original.

I tried the following, and it doesn't send a copy:

```
config.interceptor_params = {bcc: {email: "mycopy@example.com"}}
```

I also tried the following, based on the Mandrill API docs:

```
config.interceptor_params = {bcc_address: "mycopy@example.com"}
```

No luck with that, either. Only the original email is sent out.

Any solutions? Is this a Mandrill_Mailer issue, or a Mandrill API issue, or a misconfiguration on my part?

**Comment by renz45** [comment]
It looks like the issue is that we're only doing a shallow merge on the message params hash, so the value of bcc is being replaced by the interceptor. https://github.com/renz45/mandrill_mailer/blob/master/lib/mandrill_mailer/template_mailer.rb#L198

This should probably be a bit more intelligent and know how to merge values that are arrays and hashes, although it also needs to know what parameters can optionally be an array of hashes as well.

I was thinking about taking a look at some stuff this Friday, including the other issue ticket in the queue. I no longer have access to a mandrill account, but if you didn't mind I'll see what I can do with just specs and the api docs.

**Comment by xtagon** [comment]
@renz45 If it would help, I can give you a Mandrill API key to use for testing. That would be no problem.

**Comment by renz45** [comment]
Awesome! I might take you up on that offer, we'll see if the specs are enough. :) I'll let you  know.

**Comment by renz45** [comment]
I worked on this a bit Friday. Mostly cleaning up code resulting from various pull requests, getting rid of a lot of duplication and supporting all available attributes from the messages api. It needs a few more specs and then I'll add in the interceptor stuff.

Going to try to finish it this week.

**Comment by renz45** [comment]
Err its on the (v1 branch)[https://github.com/renz45/mandrill_mailer/tree/v1] I'm bumping it to v1, since I think I'll also bump the api gem to the newest version as well. Changing out interceptors work could also impact the api in minor ways so its probably safer.

**Comment by xtagon** [comment]
Thanks, just let me know when it's released. Will it be possible to set more than one BCC recipient (as well as the original recipient)? It wasn't clear in the Mandrill API docs whether this is possible, looks like it takes just one bcc_address.

**Comment by renz45** [comment]
Hi again, Sorry about the longish wait. I had some stuff come up that was sucking all my time. I just released version 1.0, and with it new interceptor functionality. I decided to go with something that will be simpler to maintain in them gem. You use a proc now as the interceptor which gets passed the mailer args. You can modify these are args to your hearts desire and we don't have to figure out any complex api specific merge logic. Checkout the readme for an example, I used your multiple bcc addresses as the example. :)

**Comment by xtagon** [comment]
Works perfectly. Thanks @renz45!
